backend/dataall/modules/datasets_base/db/dataset_models.py

```diff
@@ -0,0 +1,42 @@
+from sqlalchemy import Boolean, Column, String, Text, ForeignKey
+from sqlalchemy.dialects.postgresql import JSON, ARRAY
+from sqlalchemy.orm import query_expression
+from dataall.base.db import Base, Resource, utils
+from dataall.modules.datasets_base.services.datasets_enums import ConfidentialityClassification, Language
+
+
+class DatasetBase(Resource, Base):
+    __tablename__ = 'dataset'
+    environmentUri = Column(String, ForeignKey('environment.environmentUri'), nullable=False)
+    organizationUri = Column(String, nullable=False)
+    datasetUri = Column(String, primary_key=True, default=utils.uuid('dataset'))
+    region = Column(String, default='eu-west-1')
+    AwsAccountId = Column(String, nullable=False)
+    userRoleForDataset = query_expression()
+    userRoleInEnvironment = query_expression()
+    isPublishedInEnvironment = query_expression()
+    projectPermission = query_expression()
+    language = Column(String, nullable=False, default=Language.English.value)
+    topics = Column(ARRAY(String), nullable=True)
+    confidentiality = Column(String, nullable=False, default=ConfidentialityClassification.Unclassified.value)
+    tags = Column(ARRAY(String))
+    inProject = query_expression()
+
+    businessOwnerEmail = Column(String, nullable=True)
+    businessOwnerDelegationEmails = Column(ARRAY(String), nullable=True)
+    stewards = Column(String, nullable=True)
+
+    SamlAdminGroupName = Column(String, nullable=True)
+    autoApprovalEnabled = Column(Boolean, default=False)
+
+    datasetType = Column(String, nullable=False)
```

Comment:
any need for `datasetType` to be an enum or have a default of S3>

Reply:
I opt not to use sa.Enums as type for the column because I remember running into a similar situation in the past; I checked and I found that when I refactored notifications, I transformed one column from enum to string. To be honest I just had a soft memory of enums in columns = problems; but I think that it only applies to cases in which the list in the Enum varies often or if we get the values from another module, because that results in circular depdendencies. In short, I will add it